### DISCUSSION TITLE
How can i download images from Rest API?

### DISCUSSION THREAD

**Discussion by Uranbold** [vercel/next.js#17624] [Help]
Hello,

I'm using the getStaticProps + revalidate. How can I download the images from the Rest API? My REST is JSON and it's uploaded on a different domain. What's a good way to start?

Thanks.

**Comment by leerob** [comment]
I'm guessing you don't want to "download" images but fetch them inside `getStaticProps`. You can call your API from inside here to retrieve your images and forward them as props to your component. Unless I'm misunderstanding your question.

**Reply by Uranbold** [reply]
Yeah, I'm using the getStaticProps. All images are on the Rest API. Any example for catching them and download it 👍  I'm totally newbie on the react. Do you have any good reading or a starter?

Thanks.

**Comment by rokinsky** [comment]
Assuming you will have URLs to images and you can use those links in rendering.

```js
const Images = ({ images }) => {
  return (
    <>
    {images?.map((image) => {
       return <img src={image.url} />;
    })}
    </>
  );
};

export const getStaticProps = async () => {
  const res = await fetch("http://example.com/api/images");
  const images = await res.json();

  return {
    props: {
      images
    },
    revalidate: 60,
  };
};

export default Images;
```
